presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -319,6 +320,9 @@ public void onLifespanFinished(Iterable<Lifespan> newlyCompletedDriverGroups)
 
         public SettableFuture schedule(SourcePartitionedScheduler scheduler)
         {
+            // Return a new future even if newDriverGroupReady has not finished.
```

Comment:
Why don't we add these comments as method javadoc? Ditto for the `ListenableFuture object retention issue.` part.